### PR TITLE
Small fix for HTML Table library

### PR DESCRIPTION
**Pull request by gxgpet**
- when calling `clear()`, caption should be cleared too;
- method chaining wasn't working when calling `set_caption()` method;
- updated docs for these changes.

### CODE REVIEW

**Comment by narfbg** [comment]
In addition to my comment here - https://github.com/bcit-ci/CodeIgniter/issues/4816#issuecomment-251051151, we generally don't accept changes of multiple functionalities at once. Method chaining has nothing to do with this.

**Comment by gxgpet** [comment]
Agree. Next time I will make a PR for it.

But as for now, the devs _don't have to_ really know how to reset the caption, let's say, that we want them to use set_caption again for resetting. In this case, should they use `set_caption(FALSE)`, `set_caption('')` or `set_caption(NULL)`? That is my point.

File: user_guide_src/source/libraries/table.rst [hunk 1]
```diff
@@ -275,11 +275,12 @@ Class Reference
 		:returns:	CI_Table instance (method chaining)
 		:rtype:	CI_Table
 
-		Lets you clear the table heading and row data. If you need to show multiple tables with different data you should to call this method
+		Lets you clear the table heading, row data, and the caption. If you need to show multiple tables with different data you should to call this method
```
Comment (ValentinNikolaev) [inline user_guide_src/source/libraries/table.rst]: Please remove comma before "and"

Comment (gxgpet) [inline user_guide_src/source/libraries/table.rst]: Fixed. Thanks!

**Comment by narfbg** [comment]
@gxgpet I haven't decided whether we want this or not yet, but either way I wouldn't merge this until it is stripped down to solving only a single problem. The method chaining thing has to go, sorry - if we don't comply with our own rules, they'd be meaningless.

@jim-parry @mwhitneysdsu @lonnieezell Thoughts on the `clear()` thing?

**Comment by jim-parry** [comment]
I agree that clear() should reset the caption. That feels like an omission in the original.

**Comment by gxgpet** [comment]
@narfbg I can take out the method chaining fix from this PR and make a separate one for it, lately this day. Is that ok with you?

**Comment by narfbg** [comment]
Yep, that would work.

**Comment by mwhitneysdsu** [comment]
My only reservation about clearing the caption in `clear()` is that it's a BC break.

I think @jim-parry is right in saying it feels like an omission, and I think people use captions on tables rarely enough that it probably won't cause a lot of issues (besides, the only reason it would cause an issue would be if someone was creating a series of tables with the same caption on purpose).

I also feel that the rarity of using both captions and this library is the only reason the omission has gone unnoticed this long. It is apparent that caption support was added after the fact, and someone would have needed to be very thorough to have noticed that the caption wasn't also added to the `clear()` method (and really, looking at the PR for `set_caption()`, I don't think it would have been accepted without some changes in the ~5 years I've been following this repo, so things were a little different in 2007).

So, I guess I'm saying that I support adding the caption to the `clear()` method. The documentation isn't specific enough to give someone the expectation that the caption wouldn't also be reset, so someone would only depend on that behavior based on observing the behavior of the code (or reading the code directly).

**Comment by narfbg** [comment]
The BC break is my major concern as well. I guess we may merge, but keep it for version 3.2.0.

**Pull request merged**